packages/jaspr_lints/lib/src/assists/tree_assist.dart

```diff
@@ -0,0 +1,221 @@
+      message: 'Wrap with Builder',
+    );
+
+    cb2.addDartFileEdit((builder) {
+      builder.addReplacement(node.sourceRange, (edit) {
+        edit.write('Builder(builder: (context) sync* {\n');
+        edit.write(''.padLeft(lineIndent, ' '));
+        edit.write('  yield ');
+        edit.write(lines[0]);
+        edit.write(lines.skip(1).map((s) => '\n  $s').join());
+        edit.write(';\n${''.padLeft(lineIndent, ' ')}})');
+      });
+    });
+  }
+
+  void removeComponent(CustomLintResolver resolver, ChangeReporter reporter, Expression node, int lineIndent,
+      ArgumentList argumentList) {
+    var children = <AstNode>[
+      for (var arg in argumentList.arguments)
+        if (arg is NamedExpression)
+          if (arg.name.label.name == 'child' && isComponentType(arg.staticType))
+            arg.expression
+          else if (arg.name.label.name == 'children' && arg.expression is ListLiteral)
+            ...(arg.expression as ListLiteral).elements
+          else
+            ...[]
+        else if (arg is ListLiteral)
+          ...arg.elements,
+    ];
+
+    if (children.isEmpty) {
+      return;
+    }
+
+    var content = resolver.source.contents.data;
+
+    if (children.length == 1) {
+      final cb = reporter.createChangeBuilder(
+        priority: 3,
+        message: 'Remove this component',
+      );
+
+      cb.addDartFileEdit((builder) {
+        builder.addReplacement(node.sourceRange, (edit) {
+          var child = children.first;
+          var childIndent = getLineIndent(resolver.lineInfo, child);
+          edit.write(content.substring(child.offset, child.end).reIndent(lineIndent - childIndent, skipFirst: true));
+        });
+      });
+    } else if (node.parent is ListLiteral) {
+      final cb = reporter.createChangeBuilder(
+        priority: 3,
+        message: 'Remove this component',
+      );
+
+      cb.addDartFileEdit((builder) {
+        builder.addReplacement(node.sourceRange, (edit) {
+          for (var child in children) {
+            var childIndent = getLineIndent(resolver.lineInfo, child);
+            var source = content.substring(child.offset, child.end).reIndent(lineIndent - childIndent, skipFirst: true);
+
+            if (child != children.first) {
+              edit.write(''.padLeft(lineIndent));
+            }
+            edit.write(source);
+            if (child != children.last) {
+              edit.write(',\n');
+            }
+          }
+        });
+      });
+    } else if (node.parent is YieldStatement) {
+      final cb = reporter.createChangeBuilder(
+        priority: 3,
+        message: 'Remove this component',
+      );
+
+      cb.addDartFileEdit((builder) {
+        builder.addReplacement(node.parent!.sourceRange, (edit) {
+          for (var child in children) {
+            var childIndent = getLineIndent(resolver.lineInfo, child);
+            var source = content.substring(child.offset, child.end).reIndent(lineIndent - childIndent, skipFirst: true);
```

Comment:
I see a duplicity code in code blocks upper.

Reply:
There are subtle differences. I think its more readable this way.